src/common/cockpitwebserver.c

```diff
@@ -1028,46 +1028,60 @@ on_request_input (GObject *pollable_input,
   gsize length;
   gssize count;
 
-  length = request->buffer->len;
-  g_byte_array_set_size (request->buffer, length + 4096);
-
-  count = g_pollable_input_stream_read_nonblocking (input, request->buffer->data + length,
-                                                    4096, NULL, &error);
-  if (count < 0)
+  /* read all pending data */
```

Comment:
This should say:

/* 
 * HACK: https://gitlab.gnome.org/GNOME/glib-networking/issues/20
 * GLib incorrectly requires a looping in GPollableInputStream callback for TLS
 */